k2/python/k2/fsa.py

```diff
@@ -930,9 +930,16 @@ def invert_(self) -> 'Fsa':
             raise RuntimeError('current invert_ method only supports case '
                                'where aux_labels is a tensor')
 
+        self.__dict__['_properties'] = None
+
+        # Fix aux_labels in the final arcs to be -1, or the inverted fsa will
+        # be invalid.
+        self.aux_labels = torch.where(self.labels == -1, -1, self.aux_labels)
```

Comment:
@danpovey  Do you mean we can only change the 0 in aux_label to -1, like follows:
```
mask = torch.logical_and(self.labels == -1,  self.aux_labels == 0)
self.aux_labels = torch.where(mask, -1, self.aux_labels)
```